Nitro plugin path doesn't resolve correctly when `compatibilityVersion: 4` is set

### Environment

------------------------------
- Operating System: Windows_NT
- Node Version:     v20.10.0
- Nuxt Version:     3.12.2
- CLI Version:      3.12.0
- Nitro Version:    -
- User Config:      future, nitro, devtools
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

https://github.com/kakkokari-gtyih/nuxt-v4-directory-nitro-server-plugin-repro

### Describe the bug

When I try to use Nitro plugin following the instruction on https://nuxt.com/docs/getting-started/styling#modifying-the-rendered-head-with-a-nitro-plugin and try to simulate v4 behavior by setting `compatibilityVersion: 4`, the plugin paths are not resolving correctly and fails to run nuxt.

### Additional context

_No response_

### Logs

```shell-script
[23:20:00]  ERROR  [worker reload] [worker init] Cannot find module 'C:\app@\server\plugins\appendComment.ts' imported from C:\path_to_the_folder_with_bunch_of_node_projects\nuxt-v4-directory  at defaultResolve (node:internal/modules/esm/resolve:1121:11)
  at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:396:12)
  at ModuleLoader.resolve (node:internal/modules/esm/loader:365:25)
  at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:240:38)
  at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:85:39)
  at link (node:internal/modules/esm/module_job:84:36)
```

--- a/packages/nuxt/src/core/nitro.ts
+++ b/packages/nuxt/src/core/nitro.ts
@@ -6,7 +6,7 @@ import { createRouter as createRadixRouter, exportMatcher, toRouteMatcher } from
 import { joinURL, withTrailingSlash } from 'ufo'
 import { build, copyPublicAssets, createDevServer, createNitro, prepare, prerender, scanHandlers, writeTypes } from 'nitropack'
 import type { Nitro, NitroConfig, NitroOptions } from 'nitropack'
-import { findPath, logger, resolveIgnorePatterns, resolveNuxtModule, resolvePath } from '@nuxt/kit'
+import { findPath, logger, resolveAlias, resolveIgnorePatterns, resolveNuxtModule, resolvePath } from '@nuxt/kit'
 import escapeRE from 'escape-string-regexp'
 import { defu } from 'defu'
 import fsExtra from 'fs-extra'
@@ -217,6 +217,9 @@ export async function initNitro (nuxt: Nuxt & { _nitro?: Nitro }) {
   // Resolve user-provided paths
   nitroConfig.srcDir = resolve(nuxt.options.rootDir, nuxt.options.srcDir, nitroConfig.srcDir!)
   nitroConfig.ignore = [...(nitroConfig.ignore || []), ...resolveIgnorePatterns(nitroConfig.srcDir), `!${join(nuxt.options.buildDir, 'dist/client', nuxt.options.app.buildAssetsDir, '**/*')}`]
+
+  // Resolve aliases in user-provided input - so `~/server/test` will work
+  nitroConfig.plugins = nitroConfig.plugins?.map(plugin => plugin ? resolveAlias(plugin, nuxt.options.alias) : plugin)
 
   // Add app manifest handler and prerender configuration
   if (nuxt.options.experimental.appManifest) {

